app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Select.php

```diff
@@ -66,7 +66,7 @@ public function getHtml()
         $html = '<select name="' . $this->_getHtmlName() . '" id="' . $this->_getHtmlId() . '" class="no-changes">';
         $value = $this->getValue();
         foreach ($this->_getOptions() as $option) {
-            if (is_array($option['value'])) {
+            if (array_key_exists('value', $option) && is_array($option['value'])) {
                 $html .= '<optgroup label="' . $this->escapeHtml($option['label']) . '">';
                 foreach ($option['value'] as $subOption) {
```

Comment:
If `$option['value']` is null then using in the foreach should still cause a warning, no?